salt/modules/virt.py

```diff
@@ -1967,6 +1967,42 @@ def update(name,
                 devices_node.extend(changes[dev_type]['sorted'])
                 need_update = True
 
+    # Update the serial and the console definitions
+    if 'serial_type' in kwargs:
+        serial_node = devices_node.find('serial')
+        if serial_node and serial_node.attrib['type'] != kwargs['serial_type']:
+            serial_node.set('type', kwargs['serial_type'])
+            need_update = True
+        if serial_node is None:
+            devices_node.append('serial').set('type', kwargs['serial_type'])
+            need_update = True
```

Comment:
You got the idea, but now let's handle the possible multiple serial devices. I think it would be better to mimic the disk or interfaces code to do this since it handles lists and strives to normalize the XML elements to compare them. See the first commit of PR #56414 for a fix in that area.